source/adiabatic_conditions/initial_profile.cc

```diff
@@ -72,11 +72,14 @@ namespace aspect
           in.position[0] = representative_point;
           in.temperature[0] = temperatures[i-1];
           in.pressure[0] = pressures[i-1];
+          in.velocity[0] = Tensor <1,dim> ();
 
           for (unsigned int c=0; c<n_compositional_fields; ++c)
             in.composition[0][c] = this->get_compositional_initial_conditions().initial_composition(representative_point, c);
 
           in.strain_rate.resize(0); // we do not need the viscosity
+          in.cell = this->get_dof_handler().end(); // and do not know the cell index
```

Comment:
I think it would be better to write this as
```
  in.cell = typename DoFHandler<dim>::active_cell_iterator();
```
This simply generates an invalid iterator, no reference to a particular DoFHandler necessary.